tests/integration/test_attach_partition_distinct_expression_replicated/test.py

```diff
@@ -0,0 +1,90 @@
+import pytest
+from helpers.cluster import ClickHouseCluster
+from helpers.test_tools import assert_eq_with_retry
+
+cluster = ClickHouseCluster(__file__)
+
+replica1 = cluster.add_instance(
+    "replica1", with_zookeeper=True, main_configs=["configs/remote_servers.xml"]
+)
+replica2 = cluster.add_instance(
+    "replica2", with_zookeeper=True, main_configs=["configs/remote_servers.xml"]
+)
+
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    except Exception as ex:
+        print(ex)
+    finally:
+        cluster.shutdown()
+
+
+def create_table(node, table_name, replicated):
+    replica = node.name
+    engine = (
+        f"ReplicatedMergeTree('/clickhouse/tables/1/{table_name}', '{replica}')"
+        if replicated
+        else "MergeTree()"
```

Comment:
JFYI It's easy to write functional tests with multiple replicas, you can create them like:
```
create table t1 ... engine=ReplicatedMergeTree('/clickhouse/tables/{database}/{table_name}', '1');
create table t2 ... engine=ReplicatedMergeTree('/clickhouse/tables/{database}/{table_name}', '2');
```

Reply:
I did introduce some stateless tests for replicated merge tree: https://github.com/ClickHouse/ClickHouse/pull/39507/files#diff-2285549b3c6a98c7292c0dd01646ea60e35a6af06683f52ee9a57144ca0af100R313.

I don't quite remember the reasoning behind these integration, but I would guess at the time I was trying to validate it also worked on different servers / clusters?